src/main/java/com/thebois/models/World.java

```diff
@@ -0,0 +1,42 @@
+package com.thebois.models;
+
+import com.thebois.models.tiles.ITile;
+
+/**
+ * World creates a matrix and keeps track of all the structures and resources in the game world.
+ */
+public class World implements IFinder {
+
+    private ITile[][] worldMatrix;
+
+    private World() {
+    }
+
+    /**
+     * Initiates the world with the given size.
+     *
+     * @param worldSize The amount of tiles in length for X and Y, e.g. worldSize x worldSize.
+     */
+    public World(int worldSize) {
+        worldMatrix = new ITile[worldSize][worldSize];
+    }
+
+    /**
+     * Locates an object in the world and returns it.
+     *
+     * @return Object.
+     */
+    public Object find() {
+        return null;
+    }
+
+    /**
+     * Returns the matrix of the world.
+     *
+     * @return ITile[][]
+     */
+    public ITile[][] getWorld() {
+        return worldMatrix;
```

Comment:
No defensive copy?

Reply:
I can create a new array but not new ITiles inside it, therfore the objects inside still might be mutable. Or maybe not? 

If they are still mutable we could implement an interface for all things in the world like, ICopy, that would force all obejcts to create a defensive copy method. Then I would be sure I could implement a defensive copy to world!

Reply:
Ah, fair enough.

Yeah, having some sort of way to clone an object without keeping a reference to it would be good to have. Question is where - should you create an `IClonable<T>` interface in this pull request since you were the first to need it? That kind of makes sense. Otherwise we would have to create an entirely new PR from the boilerplate branch.

Reply:
On second thought, maybe allowing the view to keep a reference to the world tile matrix isn't that bad? Since both World and the view have the same matrix, there would be no need for the controller to keep sending new data to the world view.

Having to deep-clone the matrix every frame might be unnecessary.